modal/_asgi.py

```diff
@@ -29,7 +29,22 @@ async def fetch_data_in():
             try:
                 first_message = await asyncio.wait_for(message_gen.__anext__(), 5.0)
             except asyncio.TimeoutError:
-                await messages_to_app.put({"type": "http.disconnect"})
+                if scope["type"] == "http":
+                    await messages_to_app.put({"type": "http.response.start", "status": 502})
```

Comment:
Does this mean that we'll respond with `502` also when users take too long to send the request body?

Reply:
No, this is just for the first protocol message!

```
            # Cancel an ASGI app call if the initial message is not received within a short timeout.
            #
            # This initial message, "http.request" or "websocket.connect", should be sent
            # immediately after starting the ASGI app's function call. If it is not received, that
            # indicates a request cancellation or other abnormal circumstance.
```